src/cache.rs

```diff
@@ -782,18 +808,37 @@ impl Cache {
                 cache_err
                     .unwrap_error("cache::prepare_stdlib(): expected standard library to be parsed")
             })?;
-        Ok(())
+        let eval_env = self.mk_global_eval().unwrap();
+        Ok(GlobalEnv { eval_env, type_env })
     }
 
-    /// Generate a global environment from the list of `file_ids` corresponding to the standard
+    /// Generate a global typing environment from the list of `file_ids` corresponding to the standard
     /// library parts.
-    pub fn mk_global_env(&self) -> Result<eval::Environment, CacheError<Void>> {
-        if let Some(ids) = self.stdlib_ids.as_ref().cloned() {
-            let mut env = eval::Environment::new();
+    pub fn mk_global_types(&self) -> Result<typecheck::Environment, CacheError<Void>> {
+        let type_env = self
```

Comment:
It's call `type_env` but it looks like it's a vector containing the stdlib entries, isn't it?

Reply:
Yes it is ;)
Will rename it.